Develop

akotsoke opened the pull request.

akotsoke committed:
Produce config files for Generic plotter

akotsoke committed:
Files used by QC3 and QC4 scripts

akotsoke committed:
Produce Files for Generic Plotter

akotsoke committed:
Produce config files for generic plotter

akotsoke committed:
Produce config files for generic plotter

akotsoke committed:
Made some corrections

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+import numpy as np
+from ROOT import TGraph
+from ROOT import TMultiGraph
+import ROOT as r
```
lines 5 & 6 are unnecessary if you are importing ROOT.  You should use:

`r.TGraph( ... )`
`r.TMultiGraph( ... )`

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+
+
+
+
```
Typically we try to limit unnecessary white space in coding as it makes it harder to see all the code at once (making it harder to understand).  At most you should have a single empty line between blocks of code.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+#parser.add_argument('--test',action="store",dest="test",help="Select the type of test: QC3, QC4_HV QC4_SS")
+if len(sys.argv)==1:
+    parser.print_help()
+    sys.exit(1)
```
Please end case structures (if statements, loops, etc...) with the keyword 'pass'

For example:

```
if condition:
    some_code
    pass
```

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+		ret_int='kGreen+2'
+	
+
+	return ret_int
```
White space reduction and inserting `pass` into case structure needed

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+from ROOT import TMultiGraph
+import ROOT as r
+import argparse
+from optparse import OptionParser
```
Remove this line, you don't use optparse

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+from ROOT import TGraph
+from ROOT import TMultiGraph
+import ROOT as r
+import argparse
```
Duplicate import, remove (e.g. at line 10 you import from options.py which itself imports argparse).

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+
+#parser = argparse.ArgumentParser(add_help=True)
+#parser.add_argument('file', nargs='+')
+#parser.add_argument('--test',action="store",dest="test",help="Select the type of test: QC3, QC4_HV QC4_SS")
```
remove commented lines

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+
+
+
+
```
white space reduction

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+text_file.write("    Canv_Title_Offset_Y = '"+str(args.CanvTitleOffsetX)+"';\n")
+text_file.write("    Canv_Title_X = '"+str(args.CanvTitleX)+"';\n")
+text_file.write("    Canv_Title_Y = '"+str(args.CanvTitleY)+"';\n")
+for f in args.file:
```
use a more descriptive identifier other than `f` e.g. `file`

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+
+
+text_file = open(str(args.OutputName)+".cfg", "w")
+j=0
```
Use a more descriptive variable name

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+	text_file.write("          VAR_INDEP,VAR_DEP,VAR_INDEP_ERR,VAR_DEP_ERR\n" )
+	j=j+1
+
+	for i in range(int(args.SelectRowStart),int(args.SelectRowEnd)):
```
use a more descriptive variable than `i`

In options.py you defined the type of `SelectRowStart` and `SelectRowEnd` to be `int`.   You do not need to cast it again as an `int().` To be changed.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+		if bool(args.SetErrY)==True:
+			ErrY=ws.cell_value(i,int(args.SelectColumnErrY))
+		else:
+			ErrY=0.
```
Use of `pass` needed

bdorney replied:
Still missing some `pass` statements

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+	text_file.write("     [END_PLOT]\n" )
+
+
+
```
White space reduction needed

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+import pandas as pd
```
This import is not doing anything, you never use `pd.`

Remove

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+import xlrd
+from array import array
```
Unused import, remove

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+import pandas as pd
+import xlrd
+from array import array
+import numpy as np
```
Unused import, remove

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+
+
+
+def Color(j):
```
Use more descriptive variable name than `j`

bdorney commented on python/options.py:
```diff
@@ -0,0 +1,44 @@
+parser = argparse.ArgumentParser()
+
+parser.add_argument('file', nargs='+')
+parser.add_argument('--OutputName',action="store",dest="OutputName",default="Myconfig",help="Set Output file name, default is: Myconfig")
```
What is the type of OutputName? int, string, boolean, ...? Define the type.

Change the default to be more descriptive.  Also use "camelCase:"

https://en.wikipedia.org/wiki/Camel_case

bdorney commented on python/options.py:
```diff
@@ -0,0 +1,44 @@
+parser.add_argument('--SelectRowStart',action="store",dest="SelectRowStart",type=int,default=1,help= "Select the first Row for reading, default is 1")
+parser.add_argument('--SelectRowEnd',action="store",dest="SelectRowEnd",type=int,default=61,help= "Select the last Row for reading, default is 61")
+parser.add_argument('--AxisNDiv',action="store",dest="AxisNDiv",default="508,510",help="Set Axis Ndivisions X,Y (default is: 508,510)")
+parser.add_argument('--CanvDim',action="store",dest="CanvDim",default="1000,1000",help="Set Canvas Dimensions X,Y (default is: 1000,1000)")
```
define type of these two (lines 12-13)

bdorney commented on python/options.py:
```diff
@@ -0,0 +1,44 @@
+parser.add_argument('--CanvRangeY',action="store",dest="CanvRangeY",default="0,7",help="Set the Y-Range of the Canvas (default is 0,7)") 
+parser.add_argument('--CanvTitleOffsetX',action="store",dest="CanvTitleOffsetX",default="1.0",help="Set the X-Offset of the Canvas Title (default is 1.0)") 
+parser.add_argument('--CanvTitleOffsetY',action="store",dest="CanvTitleOffsetY",default="0.8",help="Set the Y-offset of the Canvas Title (default is 0.8)") 
+parser.add_argument('--CanvName',action="store",dest="CanvName",default="LS2_Detectors",help="Set Canvas Name, Default is: LS2_Detectors")
```
define type, lines 14-36

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+text_file.write("    Canv_Title_Offset_X = '"+str(args.CanvTitleOffsetX)+"';\n")
+text_file.write("    Canv_Title_Offset_Y = '"+str(args.CanvTitleOffsetX)+"';\n")
+text_file.write("    Canv_Title_X = '"+str(args.CanvTitleX)+"';\n")
+text_file.write("    Canv_Title_Y = '"+str(args.CanvTitleY)+"';\n")
```
If you define the type of these input arguments in options.py than casting all these variables to string will be unnecessary.  Please change.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+	workbook = xlrd.open_workbook(str(f))
+	ws = workbook.sheet_by_index(int(args.SelectSheetNum))
+	Gcharacter=int(f.index('G'))
+	legEntry=f[int(Gcharacter):int(Gcharacter+18)]
```
Gcharacter is already an `int` why are you casting it to an int a second time?  Change.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+for f in args.file:
+	
+	workbook = xlrd.open_workbook(str(f))
+	ws = workbook.sheet_by_index(int(args.SelectSheetNum))
```
Again, define type of these input variables in the options.py and then do not cast their types here.

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,28 @@
+import datetime,os,sys
+import subprocess
```
unnecessary imports, remove.

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,28 @@
+
+#print args.file
+
+cmd1 = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
```
There's only one command list here `cmd1`, why does it get a number? Change.

Hard coded path, change to get the path from the OS, for example:

https://github.com/bdorney/gem-plotting-tools/blob/master/ana_scans.py#L20

The setup script of the repository sets the `$SHELL` variable `$GEM_BASE.`  Use `$GEM_BASE` to ensure user/system independent operation.

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,28 @@
+from options import parser
+
+
+#print args.file
```
remove lines of code which have been commented out, white space reduction needed.

bdorney commented on python/options.py:
```diff
@@ -0,0 +1,44 @@
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('file', nargs='+')
```
What is this...?

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,35 @@
+import datetime,os,sys
+import subprocess
```
unnecessary imports, remove.

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,35 @@
+args = parser.parse_args()
+#print args.file
+
+cmd1 = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
```
same as above.

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,35 @@
+
+#print string
+	
+
```
same as above

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,35 @@
+
+
+
+runCommand(cmd1)
```
white space reduction needed

bdorney commented on python/QC4_SS.py:
```diff
@@ -0,0 +1,33 @@
+import datetime,os,sys
+import subprocess
```
unnecessary imports, remove.

bdorney commented on python/QC4_SS.py:
```diff
@@ -0,0 +1,33 @@
+args = parser.parse_args()
+#print args.file
+
+cmd1 = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
```
same as above

bdorney commented on python/QC4_SS.py:
```diff
@@ -0,0 +1,33 @@
+
+
+
+runCommand(cmd1)
```
white space reduction needed

bdorney commented on python/gemlogger.py:
```diff
@@ -0,0 +1,245 @@
+import logging
+import logging.config
```
The first line gives you the second line, remove.

bdorney commented on python/gemlogger.py:
```diff
@@ -0,0 +1,245 @@
+    GREEN     = '\033[92m'
+    RED       = '\033[91m'
+    BLACK     = '\033[90m'
+    END       = '\033[99m'
```
These are global variables here...why?

bdorney commented on python/gemlogger.py:
```diff
@@ -0,0 +1,245 @@
+#             "use_color":"True"
+#             },
+#         }
+#     }
```
commented code lines, remove.

bdorney replied:
Still not addressed

bdorney commented on python/gemlogger.py:
```diff
@@ -0,0 +1,245 @@
+    # stream_handler.setLevel(loglevel)
+    # stream_handler.setFormatter(color_formatter)
+
+    # logging.config.dictConfig(dictLogConfig)
```
commented code lines, remove

bdorney replied:
Still not addressed

bdorney commented on python/gemlogger.py:
```diff
@@ -0,0 +1,245 @@
+
+def gemcritical(loclogger,msg):
+    msg = colormsg(msg,loglevel=logging.CRITICAL)
+    loclogger.critical(msg)
```
lines 223-245 need `pass` statements

bdorney commented on python/wrappers.py:
```diff
@@ -0,0 +1,31 @@
+def runCommand(cmd,log=None):
```
You are not the original author, cite original location and author at the very least.

bdorney reviewed (changes requested):
No description of changes provided.  Please provide:
->Updates to CHANGELOG.txt describing changes
->Calling syntax, usage examples, and sample output
->Updates to README.txt describing python scripts and giving calling syntax and usage examples

Rename pull request to be more descriptive (what bug/issue report and/or feature request are you trying to address?

Changes to coding style needed.

Additionally if you intend to use `import argparse` instead of `from optparse import OptionParser` please make updates to all other python scripts to ensure uniformity in the repository, or switch to OptionParser (strongly preferred as other developers are more familiar with this).

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+j=0
+
+text_file.write("[BEGIN_CANVAS]\n")
+text_file.write("    Canv_Axis_NDiv = '"+str(args.AxisNDiv)+"'; #X,Y\n")
```
Lines 55 - 79, use the `\t` character to insert a tab, this will improve readability rather than manually inserting the spaces shown.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+	text_file.write("        Plot_Marker_Style = '"+str(Marker_Style)+"';\n" )
+	text_file.write("        Plot_Name = '"+str(legEntry)+"'; \n" )
+	text_file.write("        [BEGIN_DATA]\n" )
+	text_file.write("          VAR_INDEP,VAR_DEP,VAR_INDEP_ERR,VAR_DEP_ERR\n" )
```
Lines 88 - 97, again use the \t character to insert a tab, this will improve readability rather than manually inserting the spaces shown.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,131 @@
+
+
+	text_file.write("        [END_DATA]\n" )
+	text_file.write("     [END_PLOT]\n" )
```
Use of `\t` needed

bdorney reviewed (changes requested):
Follow up changes for inserting '\t' character.

akotsoke committed:
Add requested changes

akotsoke committed:
add requested changes

bdorney commented on python/QC3_Plotter.py:
```diff
@@ -0,0 +1,310 @@
+Detector_Short=array('f')
+Detector_Long=array('f')
+
+for f in args.filename:
```
Why is this file executing all this line-by-line?  The other python scripts for a given QC script make use of `python/Produce_Config_File.py` to create the config file.

The `genericPlotter` will set the style automatically, and will also do 1D fitting if the user configures the fit information correctly.

This is unnecessary code.  Please change to be more in line with other scripts.

Use the existing tools, don't reinvent them.

bdorney replied:
This comment has still not been addressed.

akotsoke replied:
This script (QC3_Plotter.py) is plotting a secondary axis for the Atm Press - time curve which generic plotter is not implemented to do yet 
So it's a temporary file

bdorney replied:
Let's change this implementation then.

The python scripts we add here should not be doing this (Otherwise you'll have to add t for every new observable instead of adding the functionality once). 

This should be included in the C++ code directly.  You can either make a method in the base class PlotterGeneric (preferred) or in the inherited classes for 1D observables (more tedious but livable).

bdorney replied:
I think if you address:

https://github.com/bdorney/CMS_GEM_Analysis_Framework/issues/13

This will solve this problem and allow for enhanced functionality.

bdorney replied:
I think if you implement the sketch of this functionality shown:

https://github.com/bdorney/CMS_GEM_Analysis_Framework/issues/13

This will solve this issue and then allow the python scripts to be simpler and allow for enhanced functionality overall.

akotsoke replied:
ok I will try do that

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,24 @@
+from options import parser
+
+args = parser.parse_args()
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
```
Do not use hard coded paths.

Use the environment variable $GEM_BASE.  Moreover why is this pointing to a directory **outside** the repository?  `Produce_Config_File.py` should be inside the repository for it to be tracked correctly.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+text_file.write("\tCanv_Title_X = '"+args.CanvTitleX+"';\n")
+text_file.write("\tCanv_Title_Y = '"+args.CanvTitleY+"';\n")
+
+for file in args.file:
```
Somewhere here you should add support for the 1D fitting that `genericPlotter` is able to do.

For example if the `Canv_Plot_Type` is a `TGraph`, `TGraphErrors`, `TH1F` 1D fitting is supported

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,24 @@
+
+cmd.append("--CanvTitleX=Divider Current #left(#muA#right)")
+cmd.append("--CanvTitleY=Applied Voltage #left(kV#right)")
+cmd.append("--LatexLine2=0.62,0.20, R_{Equiv}~=~5.000~M#Omega")
```
The value of `R_{Equiv}` here should not be hard coded.  The user should be measuring this with a multimeter at the time of the experiment (not assume it is 5.0) and then use this value here.  This should be taken from the xls sheet and plugged in here.

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,24 @@
+cmd.append("--CanvRangeX= 0,1000")
+cmd.append("--CanvRangeY= 0,7")
+cmd.append("--YaxisScale= True")
+cmd.append("--OutputName= QC4_LS2_V_vs_Imon_AllDet")
```
It would be better if the user was able to supply this from command line rather than being hard coded, or for the case of a single input file you you should taken this somehow from the input xls sheet.  E.g. take a substr from the full filename to get the detector serial number and append it here.

bdorney commented on python/QC4_SS.py:
```diff
@@ -0,0 +1,26 @@
+
+args = parser.parse_args()
+
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
```
Same comment as above.

bdorney commented on python/QC4_SS.py:
```diff
@@ -0,0 +1,26 @@
+cmd.append("--CanvRangeY= 0,50")
+cmd.append("--SetErrY=True")
+cmd.append("--SelectColumnErrY=8")
+cmd.append("--OutputName= QC4_LS2_SS_vs_Imon_AllDet")
```
Same comment as above

bdorney commented on python/QC5_Gain.py:
```diff
@@ -0,0 +1,31 @@
+from options import parser
+
+args = parser.parse_args()
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
```
Same comment as above

bdorney commented on python/options.py:
```diff
@@ -0,0 +1,50 @@
+parser.add_argument('--PlotMarkerSize',action="store",dest="PlotMarkerSize",type=str,default="0.8",help="Set Plot Marker Size (default is: 0.8)")
+
+
+
```
white space reduction

bdorney replied:
White space reduction.

Additionally you have declared the type on almost all of these variables here.  Then elsewhere in the PR you spend time (CPU) type-casting them again into the type that they already are!

This wastes CPU time and memory...remove the unnecessary type castings everywhere in the PR

bdorney replied:
Still needed

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+	return ret_int
+
+args = parser.parse_args()
+text_file = open(str(args.OutputName)+".cfg", "w")
```
Isn't this is already a string, why are you casting it again as a string?  This is unnecessary

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+j=0
+
+text_file.write("[BEGIN_CANVAS]\n")
+text_file.write("\tCanv_Axis_NDiv = '"+args.AxisNDiv+"'; #X,Y\n")
```
Concatenating strings like this in python is somewhat slow and uses additional memory.

the better way to do this would be too:

```
text_file.write("\tCanv_Axis_NDiv = '%i'; #X,Y\n"%(args.AxisNDiv))
```

Here the %i character tells that the item in paranthesis is to be treated as an `int`

The available types are:

- float `%f`
- hex `%x`
- integer `%i`
- string `%s`
- etc...

This is a more efficient way to do things, please change all instances

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+	ws = workbook.sheet_by_index(args.SelectSheetNum)
+	Gcharacter=int(f.index('G'))
+	legEntry=f[Gcharacter:Gcharacter+18]
+	Marker_Style = int(20+j)
```
Unnecessary cast, both `j` and `20` are already of type `int` remove.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+
+	text_file.write("\t\t[BEGIN_PLOT]\n" )
+	text_file.write("\t\t\tPlot_Color = '"+str(Color(j))+ "';\n" )
+	text_file.write("\t\t\tPlot_LegEntry = '"+str(legEntry)+"';\n" )
```
Are these already of type `string` based on how they are declared in `options.py`?  If so this casting is unnecessary and degrades performance.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+	text_file.write("\t\t\tPlot_Line_Style = '"+args.PlotLineStyle+"';\n" )
+	text_file.write("\t\t\tPlot_Marker_Size = '"+args.PlotMarkerSize+"';\n" )
+	text_file.write("\t\t\tPlot_Marker_Style = '"+Marker_Style+"';\n" )
+	text_file.write("\t\t\tPlot_Name = '"+str(legEntry)+"'; \n" )
```
same comment

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+	j=j+1
+
+	for i in range(args.SelectRowStart,args.SelectRowEnd):
+		if bool(args.YaxisScale)==True:
```
If `args.YaxisScale` can only take values of `0` or `1` this casting is unnecessary.  A more "pythonic" way to proceed is:

```
		if args.YaxisScale:
```

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+
+	for i in range(args.SelectRowStart,args.SelectRowEnd):
+		if bool(args.YaxisScale)==True:
+			Y= float(ws.cell_value(i,int(args.SelectColumnY))/1000)
```
You cast this to an int, and then back again to a float, why?  This uses more memory than necessary and is also unnecessary.  What's the goal here?

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+		if bool(args.YaxisScale)==True:
+			Y= float(ws.cell_value(i,int(args.SelectColumnY))/1000)
+		else:
+			Y= ws.cell_value(i,int(args.SelectColumnY))
```
Is this type casting really needed? Isn't this already an int?

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+			pass
+
+		if bool(args.SetErrX)==True:
+			ErrX=ws.cell_value(i,int(args.SelectColumnErrX))
```
same comment

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+			Y= ws.cell_value(i,int(args.SelectColumnY))
+			pass
+
+		if bool(args.SetErrX)==True:
```
Change to be more "pythonic" and remove unnecessary type casting

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+			ErrX=0.
+			pass
+
+		if bool(args.SetErrY)==True:
```
Change to be more "pythonic" and remove unnecessary type casting

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+			pass
+
+		if bool(args.SetErrY)==True:
+			ErrY=ws.cell_value(i,int(args.SelectColumnErrY))
```
Is this type casting really needed? Isn't this already an int?

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+			ErrY=0.
+			pass
+
+		X= ws.cell_value(i,int(args.SelectColumnX)) 
```
Is this type casting really needed? Isn't this already an int?

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+			pass
+
+		X= ws.cell_value(i,int(args.SelectColumnX)) 
+		text_file.write(str(X)+","+str(Y)+","+str(ErrX)+","+str(ErrY)+"\n")
```
change to not use string concatenation

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,22 @@
+from options import parser
+
+args = parser.parse_args()
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
```
same comment as above

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,22 @@
+cmd.append("--SelectRowEnd= 61")
+cmd.append("--CanvRangeX= 0,3600")
+cmd.append("--CanvRangeY= 0,35")
+cmd.append("--OutputName= QC3_LS2_Pres_vs_Time_AllDet")
```
It would be better if the user was able to supply this from command line rather than being hard coded, or for the case of a single input file you you should taken this somehow from the input xls sheet. E.g. take a substr from the full filename to get the detector serial number and append it here.

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,24 @@
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
+
+for f in args.file:
+	cmd.append(str(f))
```
unnecessary type casting?

bdorney commented on python/QC4_SS.py:
```diff
@@ -0,0 +1,26 @@
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
+
+for f in args.file:
+	cmd.append(str(f))
```
same

bdorney commented on python/QC5_Gain.py:
```diff
@@ -0,0 +1,31 @@
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
+
+for f in args.file:
+	cmd.append(str(f))
```
unnecessary type cast

bdorney commented on python/QC5_Gain.py:
```diff
@@ -0,0 +1,31 @@
+cmd.append("--CanvRangeY= 10,1000000")
+cmd.append("--SetErrY=True")
+cmd.append("--SelectColumnErrY=12")
+cmd.append("--OutputName= QC5_LS2_Gain_vs_Imon_AllDet")
```
same comment about hard coding this

bdorney commented on python/QC5_Rate.py:
```diff
@@ -0,0 +1,31 @@
+
+args = parser.parse_args()
+
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
```
same comment about this

bdorney commented on python/QC5_Rate.py:
```diff
@@ -0,0 +1,31 @@
+cmd = ["python","/Users/anastasia/desktop/QC_Plotting_Tools/Produce_Config_File.py"]
+
+for f in args.file:
+	cmd.append(str(f))
```
unnecessary type casting

bdorney commented on python/QC5_Rate.py:
```diff
@@ -0,0 +1,31 @@
+cmd.append("--CanvRangeY= 0,1800")
+cmd.append("--SetErrY=True")
+cmd.append("--SelectColumnErrY=8")
+cmd.append("--OutputName= QC5_LS2_Rate_vs_Imon_AllDet")
```
same comment

bdorney commented on python/gemlogger.py:
```diff
@@ -0,0 +1,244 @@
+        pass
+
+    # resets
+    ENDC      = '\033[0m'
```
Based on indendation is this a global variable? If so why?

bdorney reviewed (changes requested):
Still significant changes to improve readability and performance required.

akotsoke committed:
Add requested changes

akotsoke committed:
Add requested changes

akotsoke committed:
Changed the way I load the args parameters

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+    sys.exit(1)
+    pass
+
+def Color(j):
```
More descriptive name needed

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+args = parser.parse_args()
+text_file = open(args.OutputName+".cfg", "w")
+
+j=0
```
more descriptive name needed

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,118 @@
+		pass
+
+	text_file.write("\t\t\t[END_DATA]\n" )
+	text_file.write("\t\t[END_PLOT]\n" )
```
Before ending plot I think it would be useful to implement the ability for the user to specify the parameters that genericPlotter can use for 1D fitting.  I made this comment before but it has not been addressed yet

akotsoke replied:
I will add this as soon as possible..

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,31 @@
+from options import parser
+
+args = parser.parse_args()
+cmd = ["python","$GEM_BASE/python/Produce_Config_File.py"]
```
From the same file that you took `runCommand()` also take `envCheck()` and then do a check that the `$GEM_BASE` variable has been declared.

e.g.:

```
envCheck("GEM_BASE")
```

Should throw an exception if it is not declared.

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,31 @@
+args = parser.parse_args()
+cmd = ["python","$GEM_BASE/python/Produce_Config_File.py"]
+
+for f in args.file:
```
more descriptive loop index needed here

Consider using `file` instead of `f`

If `args.file` can be a list of files change to `args.fileList` for example reading this a user may not understand what is being done here

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,31 @@
+
+
+if len(args.file)==1:
+	filetype=f[int(f.index('.')):]
```
What is `f` here?  This variable name is ambiguous.

akotsoke replied:
Changed..

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,31 @@
+	filetype=f[int(f.index('.')):]
+	filename = f
+	filename= filename.replace(filetype,"")
+	cmd.append("--OutputName="+filename)
```
Are you sure this doesn't over-write the input file name?  I think this should be made to be more descriptive as requested previously.

e.g. how does the user distinguish between the input file and the produced output config file?  Generally the `.cfg` ending is not enough, consider a better way of doing this.  For example `config_X.cfg` would be better.

Apply this to all other scripts.

akotsoke replied:
Yes I am sure it doesn't over-write the input file name (tested)..It's just finds the type of the file (.xlsm for example) and replaces it with an empty string.

I have changed the names in the output files as requested.

bdorney commented on python/QC4_SS.py:
```diff
@@ -0,0 +1,35 @@
+cmd.append("--SetErrY=True")
+cmd.append("--SelectColumnErrY=8")
+
+
```
white space reduction needed

akotsoke replied:
done

bdorney commented on python/QC5_Rate.py:
```diff
@@ -0,0 +1,40 @@
+cmd.append("--SetErrY=True")
+cmd.append("--SelectColumnErrY=8")
+
+
```
white space reduction needed

akotsoke replied:
done

bdorney reviewed (changes requested):
Feedback still required.  Changes for QCX*.py scripts should be propagated to all others.

akotsoke committed:
Add requested changes

akotsoke committed:
Produce_Config_File.py: Added code to fit the plotted data (only when —Fit option is true)

Produce_Config_File.py and options.py: Changed all bool statements to strings after addressing a problem with bool(args.Fit) argument being true/false when appending to cmd

options.py: Added arguments to choose the details for the fit

QC3.py: Added Arguments to perform the Pressure-Time fit if the —Fit option is true

QC4_HV.py: Added Arguments to perform the Vmon-Imon fit if the —Fit option is true

wrappers.py: add function envCheck to check that the $GEM_BASE variable has been declared.

gemlogger.py: removed unnecessary functions and commented lines

 Please enter the commit message for your changes. Lines starting

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -86,24 +86,23 @@ def CyclicColor(DrawArg):
 
 	for row in range(args.SelectRowStart,args.SelectRowEnd):
-		if bool(args.YaxisScale):
+		if args.YaxisScale=='true':
```
This is fine, but keep in mind that this:

```
if args.YaxisScale:
```

Also works, since YaxisScale is boolean the if statement sees either 1 or 0 which evaluates to true or false.

bdorney replied:
Are you sure this works? Have you tested this?  Putting `'true'` in quotes may cause it to be interpreted as a string and not boolean.  For example from python command line:

```
>>> bTest = True
>>> if bTest:
...     print "true"
... 
true
>>> if bTest=='true':
...     print "true"
... 
>>> 
```

In the first case the if statement is executed (e.g. the word "true" is printed to terminal).  In the second case the if statement is **not** executed.

akotsoke replied:
Yes you are right.. the  'true'  is being interpreted as a string 
But as I describe in changelog  I changed all the bool statements to strings in options.py 
So the statement is working fine

bdorney replied:
This is bad design.  If the user calls the help menu:

```
If YaxisScale option is true the Y axis is plotted in kUnit. Default is: False"
```

This is reported, this leads the user to think this is boolean.  Additionally string comparison like this will fail for:

```
True
tRue
trUe
truE
TRUE
etc...
```

Change back to boolean, make the action `store_true` so the user just has to call the option if it is desired, and change the if statements to be cleaner, e.g. the first case I showed above.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -115,6 +114,23 @@ def CyclicColor(DrawArg):
+		text_file.write("\t\t\t[END_FIT]\n" )
+		pass
+
+	DrawArg=DrawArg+1
```
Get into the habit of using operators to do this, the code is cleaner when they are used:

```
>>> iValue=1
>>> print iValue
1
>>> iValue+=1
>>> print iValue
2
```

For a list of basic operators see:
https://www.tutorialspoint.com/python/python_basic_operators.htm

akotsoke replied:
ok I changed that

bdorney commented on python/QC4_HV.py:
```diff
@@ -21,7 +21,7 @@
 cmd.append("--CanvRangeX= 0,1000")
 cmd.append("--CanvRangeY= 0,7")
-cmd.append("--YaxisScale= True")
+cmd.append("--YaxisScale=true")
```
Looking at how you have defined the options menu:

```
parser.add_argument('--YaxisScale',dest="YaxisScale",action="store",type=bool, default=False,help="If YaxisScale option is True the Y axis is plotted in kUnit. Default is: False")
```

Can it be that the action is changes from 'store' to 'store_true'?  In this case then all that is needed is to call the option `--YaxisScale` rather than define the value.

Again what is motivating the choice of `argparse` over `optparser`?

akotsoke replied:
The argparse module improves on the standard library optparse module in a number of ways including:
-handling positional arguments
-supporting sub-commands
-allowing alternative option prefixes like + and /
-handling zero-or-more and one-or-more style arguments
-producing more informative usage messages
-providing a much simpler interface for custom types and actions

I just thought that since argparse is just an improvement of optparse will be better to use

bdorney replied:
This repository is already using optparser in the python tools.  Additionally the python tools for the GEM online software is all based on optparser.  To ensure easier long-term support and uniformity in GEM software please switch to optparser as requested earlier.

akotsoke committed:
include xlrd library in pip2.7 install command

bdorney commented on scripts/setup_CMS_GEM.sh:
```diff
@@ -44,7 +44,7 @@ if [[ !  -d $DIR_PIP ]]; then
 	#Installing packages
-	#pip2.7 install --user numpy scipy matplotlib ipython jupyter pandas sympy nose root_numpy
+	#pip2.7 install --user numpy scipy matplotlib ipython jupyter pandas sympy nose root_numpy xlrd
 	$DIR_PIP/bin/pip install --user numpy scipy matplotlib ipython jupyter pandas sympy nose root_numpy
```
I think you want to add that to L48 instead since L47 is commented out.  While we are at it let's remove all commented code from this script.

akotsoke replied:
yes my mistake..
I will make the corrections

bdorney reviewed (changes requested):
Cross-check if statements, use += operator, and double check setup script.

akotsoke committed:
Add xlrd library in pip install command and removed all the commented lines

akotsoke committed:
Changed DrawArg in produce_config_file.py
Reduce lines in options.py

bdorney commented:
Additionally as asked before what is the motivation of argparser over optparser?

bdorney commented:
In addition to the comments mentioned this branch needs a merge with the central repository as conflicts occur and auto-merging is not possible presently.

akotsoke committed:
->options.py: changed YaxisScale, SetErrX, SetErrY and Fit arguments store type (back to boolean)
->Produce_Config_File.py: Changed the if statement for the arguments mentioned above (boolean instead of strings)
->QC3.py: Changed the way the above arguments are appended to cmd
->QC4_*.py: Changed the way the above arguments are appended to cmd
->QC5_*.py: Changed the way the above arguments are appended to cmd

bdorney commented on python/QC3.py:
```diff
@@ -0,0 +1,43 @@
+	cmd.append("--OutputName=config_"+filename)
+	pass
+else:
+	cmd.append("--OutputName= config_QC3_LS2_Pres_vs_Time_AllDet")
```
Double check this space.  I think it should be omitted, L40, e.g.:

```
--OutputName= config_QC3_LS2_Pres_vs_Time_AllDet
```

vs.

```
--OutputName=config_QC3_LS2_Pres_vs_Time_AllDet
```

akotsoke replied:
I removed the space..

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,49 @@
+cmd.append("--SelectRowStart= 3")
+cmd.append("--SelectRowEnd= 37")
+cmd.append("--CanvRangeX= 0,1000")
+cmd.append("--CanvRangeY= 0,7")
```
Will these options be properly understood with a space character after the equals sign?

akotsoke replied:
I already checked this by  printing every time the arguments passed and they seem to be understood fine. 
Although I will remove the spaces for better uniformity.

bdorney commented on python/QC4_HV.py:
```diff
@@ -0,0 +1,49 @@
+	cmd.append("--LatexLine2=0.62,0.20, R_{Equiv}~=~%s~M#Omega"%Req)
+	pass
+else:
+	cmd.append("--OutputName= config_QC4_LS2_V_vs_Imon_AllDet")
```
same question

bdorney commented on python/QC4_SS.py:
```diff
@@ -0,0 +1,37 @@
+cmd.append("--SelectRowStart=3")
+cmd.append("--SelectRowEnd=37")
+cmd.append("--CanvRangeX= 0,1000")
+cmd.append("--CanvRangeY= 0,50")
```
same question

bdorney commented on python/QC5_Rate.py:
```diff
@@ -0,0 +1,42 @@
+cmd.append("--SelectRowStart=29")
+cmd.append("--SelectRowEnd=45")
+cmd.append("--CanvRangeX= 0,750")
+cmd.append("--CanvRangeY= 0,1800")
```
same question

bdorney commented on python/options.py:
```diff
@@ -0,0 +1,55 @@
+import argparse
```
Again, switch from argparser to optparser.  The argparser appears to be included only after python3.2.  The default version of python on slc6 is python2.6.6, on cc7 it is 2.7.5, and on lxplus we are setting it to 2.7.4 in the repository setup.  We are not supporting python 3.X as this would require changes to the other python scripts.

bdorney replied:
Let's also rename this file to not be ambiguous.  There is `$GEM_BASE/python/AnalysisOptions.py` file already.  This file should be renamed to `$GEM_BASE/python/PlotOptions.py`.  To do this try using:

git mv options.py PlotOptions.py

akotsoke replied:
ok I will do that..

bdorney commented on python/options.py:
```diff
@@ -0,0 +1,55 @@
+parser.add_argument('--LatexLine2',action="store",dest="LatexLine2",type=str,default="",help= "Add Latex line, example: 0.62,0.86, #splitline{LS2}{Detector~Production}")
+parser.add_argument('--LatexLine3',action="store",dest="LatexLine3",type=str,default="",help= "Add Latex line, example: 0.62,0.86, #splitline{LS2}{Detector~Production}")
+parser.add_argument('--LatexLine4',action="store",dest="LatexLine4",type=str,default="",help= "Add Latex line, example: 0.62,0.86, #splitline{LS2}{Detector~Production}")
+parser.add_argument('--LatexLine5',action="store",dest="LatexLine5",type=str,default="",help= "Add Latex line, example: 0.62,0.86, #splitline{LS2}{Detector~Production}")
```
Lines 16-21; This is not a good implementation.  It's tedious and limits the user to only allowing a maximum of 6 lines.  I think it would be better if you had something like this using the `append` action:

```
parser. add_option('--latexLines', action="append", type="string", dest="latexLines", default=None, help= "Add an entry to the latexLines list following syntax: 'Coord_PadX,Coord_PadY, text'", metavar="latexLines")
```

See documentation: https://docs.python.org/2/library/optparse.html#other-actions

akotsoke replied:
I have changed that..

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,137 @@
+text_file.write("\tCanv_Latex_Line = '%s';\n"%(args.LatexLine2))
+text_file.write("\tCanv_Latex_Line = '%s';\n"%(args.LatexLine3))
+text_file.write("\tCanv_Latex_Line = '%s';\n"%(args.LatexLine4))
+text_file.write("\tCanv_Latex_Line = '%s';\n"%(args.LatexLine5))
```
Lines 47-52 taking into account the feedback I have given below this should be a loop:

```
for line in options.latexLines:
    text_file.write("\tCanv_Latex_Line = '%s';\n"%s(line)
    pass
```

akotsoke replied:
done..

bdorney reviewed (changes requested):
Moving along nicely, try to incorporate also changes outlined here: https://github.com/bdorney/CMS_GEM_Analysis_Framework/issues/13

akotsoke committed:
->options.py: Switched from argparse to optparse, add append action to LatexLines option
->Produce_Config_File.py: add loop to print Latexlines from list
-> QC3.py,QC4_*.py,QC5_*.py: append LatexLines to cmd

akotsoke committed:
changed the name of options.py to PlotOptions.py

akotsoke committed:
-> README.txt: Added instructions for Produce_Config_File.py,QC3.py,QC4_*.py and QC5_*.py scripts ( 3.c.iii Analysis of QC data stored in an excel file )
-> QC3.py,QC4_*.py and QC5_*.py: Import PlotOptions instead of Options

bdorney commented:
This needs merge with central repository.  Right now automatic merging is not possible.

bdorney commented on README.txt:
```diff
@@ -706,6 +706,260 @@
             python2.7 python/computeEffCurves.py --file=config/configEffPredictor.cfg --deubg=True
 
+
+	# 3.c.iii Analysis of QC data stored in an excel file
```
Let's change the title here.  You're not really analyzing the data found in these excel files, you're plotting it.  The README should reflect this.

Additionally while these are python scripts that you have written it makes more sense to structure this under `Section 3.b. genericPlotter` after the first helper script subsection.  e.g.

```
          # 3.b.ii  Helper Script - Parsing Excel Files to genericPlotter
```

Let's move this all to start after Line #617 but before Line #618.

akotsoke replied:
ok I will make the changes..

bdorney commented on README.txt:
```diff
@@ -706,6 +706,260 @@
+	# 3.c.iii Analysis of QC data stored in an excel file
+        # --------------------------------------------------------
+	main script: Produce_Config_File.py
+	additional scripts: QC3.py, QC4_HV.py, QC4_SS.py, QC5_Gain.py, QC5_Rate.py
```
Make a separate section in the README which describes these separate scripts.  Here you have only described `Produce_Config_File.py`

bdorney commented on README.txt:
```diff
@@ -706,6 +706,260 @@
+                        if true (default) perform a fit to the TObject defined
+                        in the [BEGIN_PLOT]header this [BEGIN_FIT] header is
+                        found in
+  	--FitRange=FITRANGE   Set the Fit range, default is 0,1
```
You need to merge your branch with the central branch.  Your branch has diverged and does not include the changes.

Additionally re-write this section in [MarkDown](https://guides.github.com/features/mastering-markdown/).  Don't just dump the results of `Produce_Config_File.py -h` to the terminal.  Make them in a table as shown [here](https://github.com/bdorney/CMS_GEM_Analysis_Framework/blob/develop/README.md).

bdorney commented on README.txt:
```diff
@@ -706,6 +706,260 @@
+
+	
+	Example of use: 1) As with frameworkMain for each new shell navigate to the base directory of the repository 
+       and setup the environment via: source scripts/setup_CMS_GEM.sh
```
Show this as it is shown elsewhere in this section, as code that the user can immediately copy/paste, not buried in the text as you have it here.  See the different between what you've written and Sections 3.a & 3.b of the [README.md](https://github.com/bdorney/CMS_GEM_Analysis_Framework/blob/develop/README.md).

bdorney commented on README.txt:
```diff
@@ -706,6 +706,260 @@
+	
+	—————————————————————————————————————————————											
+	
+	Additional scripts: The additional scripts are designed to give specific instructions to the main script
```
Make this a separate subsection.

bdorney commented on README.txt:
```diff
@@ -706,6 +706,260 @@
+	--CanvRangeX=0,750
+	--CanvRangeY=0,1800
+	--SetErrY
+	--SelectColumnErrY=8
```
Formatting needed, this is not really descriptive.  See the examples in [README.md](https://github.com/bdorney/CMS_GEM_Analysis_Framework/blob/develop/README.md).

For example if you give this to a student, from **only** this document, can they execute this and produce plots?

bdorney reviewed (changes requested):
Changes to README needed to improve clarity.  This branch has diverged from central repository.  This needs to be resolved before continuing SW development.

akotsoke commented:
Hi Brian,
What do you mean when you are saying that this needs merge with central repository?

bdorney commented:
Scroll to the bottom of this page, see how it reads:
> This branch has conflicts that must be resolved
> Use the command line to resolve conflicts before continuing.
> Conflicting files
> CHANGELOG.txt
> README.txt
> scripts/setup_CMS_GEM.sh

This means the HEAD of the central branch (`develop`) and your branch cannot be automatically merged here.  This means there are significant changes that github cannot address automatically.

Moreover, please edit the title of this pull request and add a description as requested many times.

bdorney commented:
Please close this pull request and submit a new pull request to:

https://github.com/cms-gem-detqc-project/CMS_GEM_Analysis_Framework

Using select the `master` in the above repo for the destination of the PR.

bdorney commented:
When testing:

```
git remote add akotsoke https://github.com/akotsoke/CMS_GEM_Analysis_Framework.git
git checkout -b develop-akotsoke                                                  
git pull akotsoke develop                                                         
```

Conflicts exist:
```
git pull akotsoke develop
remote: Counting objects: 139, done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 139 (delta 50), reused 47 (delta 46), pack-reused 85
Receiving objects: 100% (139/139), 46.28 KiB, done.
Resolving deltas: 100% (104/104), completed with 7 local objects.
From https://github.com/akotsoke/CMS_GEM_Analysis_Framework
 * branch            develop    -> FETCH_HEAD
Auto-merging CHANGELOG.txt
CONFLICT (content): Merge conflict in CHANGELOG.txt
CONFLICT (delete/modify): README.txt deleted in HEAD and modified in 051864578756d204757a7cf7ac62805757f52a27. Version 051864578756d204757a7cf7ac62805757f52a27 of README.txt left in tree.
Auto-merging scripts/setup_CMS_GEM.sh
CONFLICT (content): Merge conflict in scripts/setup_CMS_GEM.sh
Automatic merge failed; fix conflicts and then commit the result.
```

Working on just `scripts/setup_CMS_GEM.sh`.  Fixed.

```
source scripts/setup_CMS_GEM.sh                                                   
python2.7 python/Produce_Config_File.py --file=data/v3ElectronicsTesting/GE11-X-S-CERN-0003_Spectrums_Imon669uA.xlsx --SelectSheetNum=0 --SelectColumnX=1 --SelectColumnY=2 --SelectRowStart=14 --SelectRowEnd=2061
Traceback (most recent call last):
  File "python/Produce_Config_File.py", line 47, in <module>
    for line in args.LatexLines:
TypeError: 'NoneType' object is not iterable
```

You need to add a check that this is not `None` before trying to iterate over it. Additionally this highlights the need for testing code before pushing it.

bdorney commented on python/PlotOptions.py:
```diff
@@ -0,0 +1,51 @@
+parser.add_option('--OutputName',action="store",dest="OutputName",type=str,default="Myconfig",help="Set Output file name, default is: Myconfig")
+parser.add_option('--SelectSheetNum',action="store",dest="SelectSheetNum",type=int,default=0,help="Select the Excel sheet for reading, default is: 0")
+parser.add_option('--SelectColumnX',action="store",dest="SelectColumnX",type=int,default=0,help= "Select Column number for X data, default value is 0")
+parser.add_option('--SelectColumnY',action="store",dest="SelectColumnY",type=int,default=0,help= "Select Column number for Y data, default value is 0")
```
Lines 8&9 Change the help menu here to indicate that counting start from 0 and not 1.

bdorney commented on python/PlotOptions.py:
```diff
@@ -0,0 +1,51 @@
+parser.add_option('--SelectColumnX',action="store",dest="SelectColumnX",type=int,default=0,help= "Select Column number for X data, default value is 0")
+parser.add_option('--SelectColumnY',action="store",dest="SelectColumnY",type=int,default=0,help= "Select Column number for Y data, default value is 0")
+parser.add_option('--SelectRowStart',action="store",dest="SelectRowStart",type=int,default=1,help= "Select the first Row for reading, default is 1")
+parser.add_option('--SelectRowEnd',action="store",dest="SelectRowEnd",type=int,default=61,help= "Select the last Row for reading, default is 61")
```
Lines 10&11 Change the help menu here to indicate that counting starts from 0 and not 1.

bdorney commented on python/Produce_Config_File.py:
```diff
@@ -0,0 +1,134 @@
+			pass
+
+		X= ws.cell_value(row,args.SelectColumnX) 
+		text_file.write(str(X)+","+str(Y)+","+str(ErrX)+","+str(ErrY)+"\n")
```
Add leading tabs here to improve readability:

```
text_file.write(""\t\t\t"+str(X)+","+str(Y)+","+str(ErrX)+","+str(ErrY)+"\n")
```

bdorney reviewed (changes requested):
Bug fixes needed
